lizard.py

```diff
@@ -674,20 +674,31 @@ def clang_warning_format(self):
 
 
 def print_warnings(option, scheme, warnings):
-    warning_count = 0
+    warning_count = None
     warning_nloc = 0
     warn_str = "!!!! Warnings ({0}) !!!!".format(
         ' or '.join("{0} > {1}".format(
             k, val) for k, val in option.thresholds.items()))
-    print("\n" + "=" * len(warn_str) + "\n" + warn_str)
-    print(scheme.function_info_head())
-    for warning in warnings:
-        warning_count += 1
+    for warning_count, warning in enumerate(warnings):
+        if warning_count == 0:
+            print("\n" + "=" * len(warn_str) + "\n" + warn_str)
+            print(scheme.function_info_head())
         warning_nloc += warning.nloc
         print(scheme.function_info(warning))
+    else:
+        if warning_count is None:
+            print_no_warnings(option)
+            warning_count = 0
     return warning_count, warning_nloc
```

Comment:
I think you can eliminate the ``warning_count = 0`` assignment in the else branch,
and do it in the return statement. Something like:
```python
return warning_count || warning_count + 1, warning_nloc
```
I am not sure if this works.